### ISSUE TITLE
enhanchment: maybe store truncation records differently

### ISSUE THREAD

**Issue by elcallio** [scylladb/scylla#445]
It is slightly dubious whether the system.local table is the right place to keep this (as a blob mind you).
The data is not transferable, even between instances (origin or scylla).  
Also, due to sharding, our management of this is slightly inefficient, so there is that too. 

At some point, we should evaluate if keeping this in some different, local, persisted storage would be 
better.

**Comment by avikivity** [comment]
One candidate is the commit log itself.

**Comment by elcallio** [comment]
Not really, since the data we need to store is a reference to an earlier point in the commit log (that may or may not exist when we save it). I.e. the data needs to be accessible independently of the CL.
Of course, if we don't care much about  replaying entries unnecessarily, we could just make sure truncation is a mutation in itself, thus added to the CL like everything else. Then there would be no need for truncation RP records. (Though maybe still timestamps).

**Comment by avikivity** [comment]
I think we need only worry about correctness, in the usual case the log if
not replayed at all, and when it is, there has usually been no truncation.
Only in a tiny minority of cases does this optimization have any effect.

I think by far the most natural approach is to store truncate as a regular
record (unusual in that it applies to an entire table rather than a
partition) .

**Comment by glommer** [comment]
Just found this bug, by coincidence. Recently truncation records were implicated in a series of bugs, in which they caused a storm of updates to system.local that took ages to be compacted away.

Indeed one of the solutions for that would be to just do away with their presence in system.local entirely.

**Comment by slivne** [comment]
We have fixed #4083 - adding a new truncation table - can we close this ?

**Comment by slivne** [comment]
@avikivity

**Comment by nyh** [comment]
> We have fixed #4083 - adding a new truncation table - can we close this ?

@elcallio ?
